localstack/services/cloudformation/models/apigateway.py

```diff
@@ -166,12 +168,45 @@ def _create(resource_id, resources, resource_type, func, stack_name):
             result = client.create_rest_api(**kwargs)
 
             body = props.get("Body")
-            if body:
+            s3_body_location = props.get("BodyS3Location", {})
```

Comment:
Just because an empty dictionary is also _falsy_

```suggestion
            s3_body_location = props.get("BodyS3Location")
```